packages/slonik/src/factories/createQueryValidator.ts

```diff
@@ -0,0 +1,41 @@
+import { type Query } from '../types';
+import { InputSyntaxError, InvalidInputError } from '@slonik/errors';
+import { stripSlonikPlaceholders } from '@slonik/sql-tag';
+import { parseAsync } from 'pgsql-parser';
+
+export const createQueryValidator = () => {
+  const queries: Record<string, number> = {};
+
+  return async (query: Query) => {
+    let stats = queries[query.sql];
+
+    if (stats === undefined) {
+      const ast = await parseAsync(stripSlonikPlaceholders(query.sql));
```

Comment:
We may want to consider what happens if `parseAsync` fails.

Minimally, if the query's invalid, we should make sure the error that's thrown includes all the details needed to determine the syntax issue.

However, you might also consider letting invalid queries through to the server -- one, that would ensure the syntax errors are the same as what we get today and two, that would avoid issues where the parser is not up to date and we try to parse some newer syntax.

Reply:
Added test to cover this scenario. Thank you